simpleviewinc/tap-events-force
# Zen 275 set session selected

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-275)

## Context

* a session item can be 'booked' by hitting the 'select' button
* if single user, update reducer to isSelected
  * call sessionBookingRequest - to book right away
* if multi-user, call the open modal, to open the group modal selection

## Goal

* Create the action `setSessionSelected`
* Setup the `BookingButton` component in the gridItem so we can easily test this action
   * NOTE: the bookingButton is not complete. it will be fully handled on 
        * https://jira.simpleviewtools.com/browse/ZEN-371
        * https://jira.simpleviewtools.com/browse/ZEN-268
* Fix `sessionLink` color on web (should have primary)

## Updates

* `src/actions/session/setSessionSelected.js`
   * Either books the current session or opens up group booking modal based on the attendees count

* `src/components/button/bookingButton.js`
   * initial booking button that will show up in the grid item

* `src/components/grid/gridTileContent.js`
   * added the bookingButton component

* `src/theme/dynamicClassMap.js`
   * ~`colors.color4` was undefined for some reason so the `SessionLink` text color was not getting the right primary color~
       * ~so i added the defaultColors.primary for the case that colors.color4 is undefined or null~
  * updated the setColors to use the default colors if the passed in setColors is undefined/null
## Testing

* `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-275-set-session-selected command=web`
* navigate to http://local.kegdev.xyz/
* Click on `SELECT` button on any grid item
* verify that you see group booking modal appears
* Close the modal
* Change the JSON data > `attendees` array to only contain 1 attendee
    * ![image](https://user-images.githubusercontent.com/3317835/94468721-bf794b00-0179-11eb-9556-fe4372c43a84.png)

* hit `Apply`
* Click on `SELECT` button on any grid item
* you should NOT see a modal appear, but instead see a console.log

### Booking Button
> NOTE: You wont see the booking button on mobile view because it only shows up when you expand it (details view)
*  Update any `session.capacity.remainingPlaces` to 0 > hit 'Apply'
   * ![image](https://user-images.githubusercontent.com/3317835/94469054-462e2800-017a-11eb-9c3d-db5f42cd943e.png)

* see that the corresponding session item does not have a booking button
    * ![image](https://user-images.githubusercontent.com/3317835/94469123-65c55080-017a-11eb-8c86-e26c8daedcc9.png)
